blurb/blurb.py

```diff
@@ -888,6 +878,18 @@ def init_tmp_with_template():
     path = blurb.save_next()
     git_add_files.append(path)
     flush_git_add_files()
+
+    # For convenience, write a preformatted commit template. Enable template
+    # with ``git config commit.template .git/blurb``.
+    msgfile = os.path.join(root, '.git', 'blurb')
+    if os.path.isdir(os.path.dirname(msgfile)):
+        metadata, text = blurb[0]
+        title = branch_suffix.replace('-', ' ').replace('_', ' ')
```

Comment:
I don't like this.  You are replacing _all_ dashes and _all_ underscores with spaces here.  This means that a branch called `"bpo-1234-Fix-__dir__-on-dict"` would be unhelpfully converted into the template `"    bpo-1234: Fix   dir   on dict"` with _three_ spaces on either side of `dict`.  (GitHub apparently refuses to display the three spaces, 
and collapses them to one during formatting.)

If we honor the name of the branch, we should honor the name of the branch.  There are legitimate uses for underscores and dashes, and stripping them all out is more surprising than simply leaving them there.  I realize git doesn't permit spaces in branch names, so we can't simply write a wonderful commit message in the branch name and have it emerge from the other end of the process perfectly.  Ultimately, leaving the dashes and underscores there may be a little inelegant but that's better than outright corrupted.

Please remove this line and resubmit the PR.